File: read_docx.py
import zipfile
import xml.etree.ElementTree as ET
import os
import sys

def extract_text(actual_path):
    try:
        with zipfile.ZipFile(actual_path) as docx:
            tree = ET.XML(docx.read('word/document.xml'))
            namespaces = {'w': 'http://schemas.openxmlformats.org/wordprocessingml/2006/main'}
            text = []
            for paragraph in tree.iterfind('.//w:p', namespaces):
                texts = [node.text for node in paragraph.iterfind('.//w:t', namespaces) if node.text]
                if texts:
                    text.append(''.join(texts))
                else:
                    text.append('')
            return '\n'.join(text)
    except Exception as e:
        return str(e)

files = [f for f in os.listdir('.') if f.endswith('.docx')]
for f in files:
    print(f"--- File: {f} ---")
    print(extract_text(f))
    print("\n")

File: read_docx.php
<?php
function readDocx($filename) {
    if(!file_exists($filename)) {
        return "File not found.";
    }

    $zip = new ZipArchive;
    if ($zip->open($filename) === TRUE) {
        if (($index = $zip->locateName('word/document.xml')) !== false) {
            $data = $zip->getFromIndex($index);
            $zip->close();

            $xml = new DOMDocument();
            $xml->loadXML($data);
            
            $text = '';
            $paragraphs = $xml->getElementsByTagNameNS('http://schemas.openxmlformats.org/wordprocessingml/2006/main', 'p');
            
            foreach ($paragraphs as $p) {
                $texts = $p->getElementsByTagNameNS('http://schemas.openxmlformats.org/wordprocessingml/2006/main', 't');
                $p_text = '';
                foreach ($texts as $t) {
                    $p_text .= $t->nodeValue;
                }
                $text .= $p_text . "\n";
            }
            return $text;
        }
        $zip->close();
    }
    return "Error reading zip or document.xml not found.";
}

$files = ["public/uploads/templates/leave_vacation_template.docx"];
$out = "";
foreach ($files as $f) {
    if(!file_exists($f)) {
        $out .= "$f not found\n";
    }
    $out .= "--- File: $f ---\n";
    $out .= readDocx($f);
    $out .= "\n\n";
}
file_put_contents('scratch/vacation_text.txt', $out);
?>
